KMaina/Store-Manager-API
# #161280197 Build out REST Endpoints for Users Resource

#### What does this PR do?
Allow a user regsiter in the system and view all users
#### Description of Task to be completed?

Have the following endpoints working

`POST /api/v1/auth/signup`

`GET /api/v1/users`

`GET /api/v1/users/<int:user_id>`

#### How should this be manually tested?

- clone the repo locally

- cd into the Store-Manager-API folder

- run `git pull https://github.com/KMaina/Store-Manager-API.git ft-user-registration-endpoint-161280197`

- in a terminal run the command `pytest`

- ensure that the following environment variables are set

```
FLASK_ENV=development
FLASK_APP=run.py
```
- in a terminal run flask run to start the server

- in Postman use the following JSON body to add a product

```
{
   "name" : "Ken",
   "password" : "pass1234",
   "confirm" : "pass1234"
}
```
#### Any background context you want to provide?
This is all functionality for the users endpoint
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/161280197
